### PR TITLE
Keep AI panel open when workspace chooser opens

### PR DESCRIPTION
**Pull request by jurgenwerk**
After closing the last card on the stack, the AI panel now stays open. 

https://github.com/user-attachments/assets/54c2bd27-5b81-401e-bda7-a92066ee0ab4

### CODE REVIEW

**Comment by FadhlanR** [comment]
The AI panel appears inconsistent in the workspace chooser. If the user toggles to the workspace chooser from interact mode, the AI panel is displayed. However, if the user toggles to the workspace chooser from code mode, the AI panel is hidden. Is this the expected behavior?

**Review by lukemelia — APPROVED** [review]

**Review by burieberry — COMMENTED** [review]
On staging and prod previews, I came across this strange behavior when I open the ai assistant and send a message, everything slides up. When I refresh it, it looks fine but then open the chat and everything slides up again. Are you able to replicate this?
workspace-chooser

**Comment by habdelra** [comment]
> On staging and prod previews, I came across this strange behavior when I open the ai assistant and send a message, everything slides up. When I refresh it, it looks fine but then open the chat and everything slides up again. Are you able to replicate this? workspace-chooser

ive seen this kind of thing in the past when we are using visibility:hidden instead of display:none. the AI panel has a scrollTo to scroll to the bottom of the chat when new messages are received. scrollTo acts really wonky when visbility:hidden starts affecting the geometry of the page. In the past we used to see this when because the search panel was using a visibility:hidden when it was dismissed. there is a chance something else was added to the DOM that had nothing to do with AI chat, but still effected the page geometry by using a hidden item. The issue is probably unrelated to this PR and likely has been lurking for some time

**Comment by jurgenwerk** [comment]
@FadhlanR perhaps the title of this PR is misleading - the intended behaviour is that the AI panel is opened in the workspace chooser only if it was opened prior to opening the workspace chooser. This can currently happen only in interact mode but not in code mode. So I think this is expected behavior.

**Comment by FadhlanR** [comment]
> @FadhlanR perhaps the title of this PR is misleading - the intended behaviour is that the AI panel is opened in the workspace chooser only if it was opened prior to opening the workspace chooser. This can currently happen only in interact mode but not in code mode. So I think this is expected behavior.

Screenshot 2024-10-08 at 15 02 57

How about this button to open AI panel ? It appears when I first time open the preview app.

**Comment by jurgenwerk** [comment]
@burieberry I was able to reproduce now. It's actually easy to reproduce -  it happens when you are in the chooser and try to open the AI panel and then the layout breaks in ways you described. But if you hide the ai panel element in the console and show it again, then it's displayed correctly. So I think this really has something to do with calculations as @habdelra pointed out. Maybe the chooser should be displayed absolutely on the top so that it doesn't interfere with what is under it. Will try something here.

**Comment by jurgenwerk** [comment]
@FadhlanR let's ask @lukemelia: does it make sense to have the AI panel button in the workspace chooser? I think it probably does because if you are able to close it then it makes sense to also be able to open it back again

**Comment by jurgenwerk** [comment]
@burieberry are you able to reproduce the problems in the pr preview link (make sure to hard refresh the page)? I think https://github.com/cardstack/boxel/pull/1657/commits/c4c22895e7221d31b5c9d17b1cd0d6799b5ef664 fixed it

File: packages/host/app/components/operator-mode/workspace-chooser/index.gts [hunk 1]
```diff
@@ -61,11 +61,13 @@ export default class WorkspaceChooser extends Component<Signature> {
...
         opacity: 0;
-        position: relative;
+        position: absolute;
         background-color: var(--boxel-700);
-        height: 100vh;
+        height: 100%;
+        width: 100%;
         padding: 5.5rem 0 5.5rem 11.5rem;
-        animation: fadeIn 1s ease-in forwards;
+        animation: fadeIn 0.5s ease-in forwards;
```
Comment (jurgenwerk) [inline packages/host/app/components/operator-mode/workspace-chooser/index.gts:69]: I think it looks nicer if the animation isn't a whole second long, I reduced it to 0.5

**Comment by lukemelia** [comment]
> @FadhlanR let's ask @lukemelia: does it make sense to have the AI panel button in the workspace chooser? I think it probably does because if you are able to close it then it makes sense to also be able to open it back again

I agree

**Review by FadhlanR — APPROVED** [review]

**Comment by habdelra** [comment]
> @burieberry I was able to reproduce now. It's actually easy to reproduce - it happens when you are in the chooser and try to open the AI panel and then the layout breaks in ways you described. But if you hide the ai panel element in the console and show it again, then it's displayed correctly. So I think this really has something to do with calculations as @habdelra pointed out. Maybe the chooser should be displayed absolutely on the top so that it doesn't interfere with what is under it. Will try something here.

ah, I hadn't thought about that. yeah, likely the geometry of the workspace chooser is effecting the scrollTo calculation. Specifically if the workspace chooser has elements that overflow. scrollTo is really annoying in that overflow:hidden has no effect on the height that scrollTo determines is the bottom of the window. So you need to be really careful with the geometry. Turn off any overflow:hidden in the workspace chooser to see the world like scrollTo sees it.

**Comment by jurgenwerk** [comment]
I spent a couple of minutes interacting with the chooser and ai panel in all sorts of ways and couldn't find any layout issues anymore.

**Pull request merged**